scipy/stats/_binned_statistic.py

```diff
@@ -294,6 +294,7 @@ def binned_statistic_2d(x, y, values, statistic='mean',
     `expand_binnumbers` argument. If 'False' (default): The returned
     `binnumber` is a shape (N,) array of linearized indices mapping each
     element of `sample` to its corresponding bin (using row-major ordering).
+    Note that the array is padded for under- and overflow.
```

Comment:
I'm not sure how to interpret this. What do you mean exactly, or what problem with under/overflow are you seeing?

Reply:
The problem I had and also some colleagues I talked to, is in understanding the output bin numbers as e.g. for `ret.binnumbers` in the [readme example](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.binned_statistic_2d.html) with the `expand_binnumbers=False` option. It took me very long to figure out the numbering of the linearized 2D bin array as it adds over and underflow bins on the edges. It was only until I stumbled over this [post](https://stackoverflow.com/questions/63275441/can-i-get-binned-statistic-2d-to-return-bin-numbers-for-only-bins-in-range) how to read the `ret.binnumber` output. I would have appreciated myself reading something like this that helps you figuring it out a lot. It can also be phrased differently, just a note on that would be great I think.

Reply:
The SO post is useful, thanks. I would suggest a more elaborate explanation about the number of bins and that the first and last bin for each dimension are for capturing values outside of the bounds. Because this one sentence you added is very cryptic - and over/underflow are not the correct terms (that has more to do with floating-point numerics).